lisa/sut_orchestrator/aws/common.py

```diff
@@ -0,0 +1,250 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import re
+import sys
+from dataclasses import InitVar, dataclass, field
+from typing import Any, Dict, List, Optional, Union
+
+from dataclasses_json import dataclass_json
+from marshmallow import validate
+
+from lisa import schema
+from lisa.environment import Environment
+from lisa.node import Node
+from lisa.util import LisaException, constants, field_metadata
+from lisa.util.parallel import check_cancelled
+from lisa.util.perf_timer import create_timer
+
+
+@dataclass
+class EnvironmentContext:
+    key_pair_name: str = ""
+    security_group_name: str = ""
+    security_group_id: str = ""
+    security_group_is_created: bool = False
+
+
+@dataclass
+class NodeContext:
+    intsance_id: str = ""
+    vm_name: str = ""
+    username: str = ""
+    password: str = ""
+    private_key_file: str = ""
+
+
+@dataclass_json()
+@dataclass
+class AwsVmMarketplaceSchema:
+    # amazon/AWS Deep Learning AMI GPU CUDA 11.2.1 (Ubuntu 20.04) 20220208
+    imageid: str = "ami-097324d9d7113bccb"
+
+
+@dataclass_json()
+@dataclass
+class AwsNodeSchema:
+    name: str = ""
+    vm_size: str = ""
+    location: str = ""
+
+    marketplace_raw: Optional[Union[Dict[Any, Any], str]] = field(
+        default=None, metadata=field_metadata(data_key="marketplace")
+    )
+
+    nic_count: int = 1
+    enable_sriov: bool = False
+    data_disk_count: int = 0
+    data_disk_caching_type: str = field(
+        default=constants.DATADISK_CACHING_TYPE_NONE,
+        metadata=field_metadata(
+            validate=validate.OneOf(
+                [
+                    constants.DATADISK_CACHING_TYPE_NONE,
+                    constants.DATADISK_CACHING_TYPE_READONLY,
+                    constants.DATADISK_CACHING_TYPE_READYWRITE,
+                ]
+            )
+        ),
+    )
+    data_disk_iops: int = 500
+    data_disk_size: int = 32
+    disk_type: str = ""
+
+    # for marketplace image, which need to accept terms
+    _marketplace: InitVar[Optional[AwsVmMarketplaceSchema]] = None
+
+    @property
+    def marketplace(self) -> Optional[AwsVmMarketplaceSchema]:
```

Comment:
What's the format of marketplace image id? If it's not complex as Azure, it doesn't need this logic.

Reply:
Updated, thanks